app/src/api/dataSource/ClientApiDataSource.ts

```diff
@@ -54,6 +62,13 @@ export class ClientApiDataSource implements ClientApi {
       JSON.stringify(params),
     );
 
+    const publicKey = getExecutorPublicKey();
```

Comment:
This should behave the same as client key -> if it is not present then logout user (later send him to identity selector)

Reply:
This is handled now with [this PR](https://github.com/calimero-network/core/pull/616). The public key will always exist in this case unless user clears this storage item manually. 
The identity is only set is login was successful so if no public key user was never login in the first place.